crates/sui-framework/packages/sui-framework/sources/metadata_value.move

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Defines the `MetadataValue` enum which represents common value types which
+/// are used to describe metadata.
+module sui::metadata_value;
```

Comment:
This feels like a really strange place for this concept and an even stranger name. But Maybe I'm missing something.
Why not just add a something like `std::bcs::Tagged` where `public struct Tagged has copy, drop, store { ty: TypeName, bytes: vector<u8> }`